packages/hooks/src/test/index.test.js

```diff
@@ -620,3 +628,30 @@ test( 'Test `this` context via composition', () => {
 	const testObject2 = {};
 	Object.assign( testObject2, createHooks() );
 } );
+
+// Test adding a hook triggers a hookAdded action.
+test( 'adding a hook triggers a hookAdded action passing all callback details', () => {
+	const testObject = {};
+
+	addAction( 'hookAdded', 'my_callback', handle_hook_added );
```

Comment:
Another way of testing it is to pass Jest spy instead of `handle_hook_added` callback and verify if spy was called with the same set of data:

```js
const hookAddedSpy = jest.fn();

...

expect( hookAddedSpy ).toHaveBeenCalledWith( 'testHook', 'my_callback2', action_a, 9 );
```

Reply:
nice, that is much cleaner. I haven't don't much test writing in Jest, thanks for pointing that out.

Reply:
No worries. I'm still learning Jest, it has really well-though API.